validate/validate.go

```diff
@@ -335,6 +336,24 @@ func (v *Validator) CheckMounts() (msgs []string) {
 	return
 }
 
+func (v *Validator) CheckOS() (msgs []string) {
+	logrus.Debugf("check os")
+
+	if v.spec.Platform.OS != "linux" {
+		if v.spec.Linux != nil {
+			msgs = append(msgs, fmt.Sprintf("When system is not linux, Linux field should be empty"))
```

Comment:
This is not supported by any spec wording, which [only SHOULDs the *setting* (and says nothing about unsetting) the platform-specific sections based on `platform.os`][1].

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc4/config.md#platform-specific-configuration